src/main/scala/com/amazon/deequ/analyzers/Distance.scala

```diff
@@ -40,32 +52,198 @@ object Distance {
+   * @param absThresholdYates            Yates absolute threshold for tables larger than 2x2
+   * @param percThresholdYates           Yates percentage of categories that can be below threshold for tables larger than 2x2
+   * @param absThresholdCochran          Cochran absolute threshold for 2x2 tables
+   * @return (sample, expected)          returns the two regrouped mappings
+   *
+   */
+  private[this] def regroupCategories(
+    sample: scala.collection.mutable.Map[String, Double],
+    expected: scala.collection.mutable.Map[String, Double],
+    absThresholdYates: Integer = 5,
+    percThresholdYates: Double = 0.2,
+    absThresholdCochran: Integer = 10)
+    : (scala.collection.mutable.Map[String, Double], scala.collection.mutable.Map[String, Double]) = {
+
+    // If number of categories is below return original mappings
+    if (expected.keySet.size < 2) {
+      (sample, expected)
+    } else {
+      // Determine thresholds depending on dimensions of mapping (2x2 tables use Cochran, all other tables Yates thresholds)
+      var absThresholdPerColumn : Integer = absThresholdCochran
+      var maxNbColumnsBelowThreshold: Integer = 0
+      if (expected.keySet.size > 2) {
+        absThresholdPerColumn = absThresholdYates
+        maxNbColumnsBelowThreshold = (percThresholdYates * expected.keySet.size).toInt
       }
-      val combinedKeys = sample1.keySet.union(sample2.keySet)
-      var linfSimple = 0.0
+      // Count number of categories below threshold
+      val nbExpectedColumnsBelowThreshold = expected.filter(e => e._2 < absThresholdPerColumn).keySet.size
 
-      combinedKeys.foreach { key =>
-        val cdf1 = sample1.getOrElse(key, 0L) / n
-        val cdf2 = sample2.getOrElse(key, 0L) / m
-        val cdfDiff = Math.abs(cdf1 - cdf2)
-        linfSimple = Math.max(linfSimple, cdfDiff)
+      // If the number of categories below threshold exceeds the authorized maximum, small categories are regrouped until valid
+      if (nbExpectedColumnsBelowThreshold > maxNbColumnsBelowThreshold){
+
+        // Identified key that holds minimum value
+        val expectedMin: (String, Double) = expected.minBy(e => e._2)
+        val sampleMinValue : Double = sample.getOrElse(expectedMin._1, 0)
+
+        // Remove smallest category
+        expected.remove(expectedMin._1)
+        sample.remove(expectedMin._1)
+
+        // Add value of smallest category to second smallest category
+        val expectedSecondMin = expected.minBy(e => e._2)
+        val sampleSecondMinValue : Double = sample.getOrElse(expectedSecondMin._1, 0)
+
+        expected.update(expectedSecondMin._1, expectedSecondMin._2 + expectedMin._2 )
+        sample.update(expectedSecondMin._1, sampleMinValue + sampleSecondMinValue )
+
+        // Recursively call function until mappings are valid
+        regroupCategories(sample, expected, absThresholdYates, percThresholdYates, absThresholdCochran)
```

Comment:
Is there any edge case where we end up in a recursive loop?

Reply:
yes should be sufficient, the stopping condition is the number of keys  being below a threshold, before the recursive call is made a key is always removed. Therefore the stopping condition will be always met